helm autodiscover crash when you set ignores

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When you set any ignore setting updatecli crash

### Expected Behavior

process the ignore settings

### Steps To Reproduce

1. Create a GitHub repo with a folder with Helm Charts
2. create the following manifest
```yaml
---
# yaml-language-server: $schema=https://www.updatecli.io/schema/latest/config.json
name: "Helm autodiscovery Docker containers"
scms:
  githubConfig:
    kind: "github"
    spec:
      user: '{{ requiredEnv "GIT_USER" }}'
      email: '{{ requiredEnv "GIT_EMAIL" }}'
      owner: "{{ requiredEnv "GITHUB_OWNER" }}"
      repository: "{{ requiredEnv "GITHUB_REPO" }}"
      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "GIT_USER" }}'
      branch: "main"
actions:
  updateCharts:
    kind: github/pullrequest
    scmid: githubConfig
    spec:
      mergemethod: "squash"
      automerge: true
      labels:
        - "dependencies"

autodiscovery:
  scmid: githubConfig
  actionid: updateCharts
  crawlers:
    helm:
      rootdir: helm-charts
      digest: true
      ignorechartdependency: true
      skippackaging: true
      versionincrement: patch
      ignore:
        - path: "helm-charts/hello"
# I tested also these configs
#        - containers:
#            "my-helm-chart": ">0.0.1"
#        - dependencies:
#           "my-helm-chart": ""
``` 

3. run updatecli

```shell
 updatecli diff --config autodiscover.yml
```

```log
HELM
=====
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xf0d143c]

goroutine 1 [running]:
github.com/updatecli/updatecli/pkg/plugins/autodiscovery/helm.Helm.discoverHelmContainerManifests({0x1, {0x0, 0xc000e1e5d8, 0x0, 0x1, {0xc000e0e760, 0x1, 0x1}, {0xc000baf3a0, 0xb}, ...}, ...})
        /home/runner/work/updatecli/updatecli/pkg/plugins/autodiscovery/helm/container.go:176 +0xb9c
github.com/updatecli/updatecli/pkg/plugins/autodiscovery/helm.Helm.DiscoverManifests({0x1, {0x0, 0xc000e1e5d8, 0x0, 0x1, {0xc000e0e760, 0x1, 0x1}, {0xc000baf3a0, 0xb}, ...}, ...})
        /home/runner/work/updatecli/updatecli/pkg/plugins/autodiscovery/helm/main.go:140 +0x219
github.com/updatecli/updatecli/pkg/core/pipeline/autodiscovery.(*AutoDiscovery).Run(0xc00091a8a0?)
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/autodiscovery/main.go:363 +0x76
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).LoadAutoDiscovery(0x136416e0, 0x0?)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/autodiscovery.go:109 +0x653
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).Prepare(0x136416e0)
        /home/runner/work/updatecli/updatecli/pkg/core/engine/prepare.go:41 +0x15f
github.com/updatecli/updatecli/cmd.run({0x10734d9a?, 0x0?})
        /home/runner/work/updatecli/updatecli/cmd/root.go:126 +0x494
github.com/updatecli/updatecli/cmd.init.func9(0x135f4e20?, {0xc000bbe500?, 0x0?, 0x2?})
        /home/runner/work/updatecli/updatecli/cmd/diff.go:38 +0x247
github.com/spf13/cobra.(*Command).execute(0x135f4e20, {0xc000bbe4e0, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x135f3720)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/updatecli/updatecli/cmd.Execute()
        /home/runner/work/updatecli/updatecli/cmd/root.go:50 +0x5b
main.main()
        /home/runner/work/updatecli/updatecli/main.go:8 +0xf
```

### Environment

```markdown
- OS: Ubuntu 22.04 (GitHub runner) / macOS 15
- updatecli: 0.91.0
```


### Pipeline Graph

```mermaid
```


### Anything else?

_No response_

--- a/pkg/plugins/autodiscovery/helm/container.go
+++ b/pkg/plugins/autodiscovery/helm/container.go
@@ -150,10 +150,28 @@ func (h Helm) discoverHelmContainerManifests() ([][]byte, error) {
 				}, "/")
 			}
 
+			imageTag := image.tag
+
 			imageSourceSlug := strings.ReplaceAll(imageName, "/", "_")
 
+			// Test if the ignore rule based on path is respected
+			if len(h.spec.Ignore) > 0 {
+				if h.spec.Ignore.isMatchingRules(h.rootDir, chartRelativeMetadataPath, "", "", imageName, imageTag) {
+					logrus.Debugf("Ignoring container version update from file %q, as matching ignore rule(s)\n", relativeFoundValueFile)
+					continue
+				}
+			}
+
+			// Test if the only rule based on path is respected
+			if len(h.spec.Only) > 0 {
+				if !h.spec.Only.isMatchingRules(h.rootDir, chartRelativeMetadataPath, "", "", imageName, imageTag) {
+					logrus.Debugf("Ignoring container version update from %q, as not matching only rule(s)\n", relativeFoundValueFile)
+					continue
+				}
+			}
+
 			// Try to be smart by detecting the best versionfilter
-			sourceSpec := dockerimage.NewDockerImageSpecFromImage(imageName, image.tag, h.spec.Auths)
+			sourceSpec := dockerimage.NewDockerImageSpecFromImage(imageName, imageTag, h.spec.Auths)
 
 			versionFilterKind := h.versionFilter.Kind
 			versionFilterPattern := h.versionFilter.Pattern
@@ -163,6 +181,8 @@ func (h Helm) discoverHelmContainerManifests() ([][]byte, error) {
 				versionFilterKind = sourceSpec.VersionFilter.Kind
 				versionFilterPattern = sourceSpec.VersionFilter.Pattern
 				tagFilter = sourceSpec.TagFilter
+				imageName = sourceSpec.Image
+				imageTag = sourceSpec.Tag
 			}
 
 			// If a versionfilter is specified in the manifest then we want to be sure that it takes precedence
@@ -175,22 +195,6 @@ func (h Helm) discoverHelmContainerManifests() ([][]byte, error) {
 					if sourceSpec != nil {
 						sourceSpec.VersionFilter.Pattern = "*"
 					}
-				}
-			}
-
-			// Test if the ignore rule based on path is respected
-			if len(h.spec.Ignore) > 0 && sourceSpec != nil {
-				if h.spec.Ignore.isMatchingRules(h.rootDir, chartRelativeMetadataPath, "", "", sourceSpec.Image, sourceSpec.Tag) {
-					logrus.Debugf("Ignoring container version update from file %q, as matching ignore rule(s)\n", relativeFoundValueFile)
-					continue
-				}
-			}
-
-			// Test if the only rule based on path is respected
-			if len(h.spec.Only) > 0 && sourceSpec != nil {
-				if !h.spec.Only.isMatchingRules(h.rootDir, chartRelativeMetadataPath, "", "", sourceSpec.Image, sourceSpec.Tag) {
-					logrus.Debugf("Ignoring container version update from %q, as not matching only rule(s)\n", relativeFoundValueFile)
-					continue
 				}
 			}
 
@@ -243,7 +247,7 @@ func (h Helm) discoverHelmContainerManifests() ([][]byte, error) {
 				ScmID                       string
 			}{
 				ImageName:                   imageName,
-				ImageTag:                    image.tag,
+				ImageTag:                    imageTag,
 				ChartName:                   chartName,
 				HasRegistry:                 image.registry != "",
 				ConditionRegistryID:         imageSourceSlug + "-registry",

--- a/pkg/plugins/autodiscovery/helm/main_test.go
+++ b/pkg/plugins/autodiscovery/helm/main_test.go
@@ -50,6 +50,7 @@ targets:
       file: 'Chart.yaml'
       key: '$.dependencies[0].version'
       name: 'epinio'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'minio'
 `,
@@ -81,6 +82,7 @@ targets:
       file: 'Chart.yaml'
       key: '$.dependencies[1].version'
       name: 'epinio'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'kubed'
 `,
@@ -112,6 +114,7 @@ targets:
       file: 'Chart.yaml'
       key: '$.dependencies[2].version'
       name: 'epinio'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'epinio-ui'
 `,
@@ -143,6 +146,7 @@ targets:
       file: 'values.yaml'
       name: 'epinio'
       key: '$.image.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'splatform_epinio-server'
 `,
@@ -174,6 +178,7 @@ targets:
       file: 'values.yaml'
       name: 'epinio'
       key: '$.images.ui.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'epinioteam_epinio-ui-qa'
 `},
@@ -218,6 +223,7 @@ targets:
       file: 'values.yaml'
       name: 'sample'
       key: '$.image.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'epinio_epinio-server-digest'
 `,
@@ -257,6 +263,7 @@ targets:
       file: 'values.yaml'
       name: 'sample'
       key: '$.images.ui.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'epinio_epinio-ui-digest'
 `},
@@ -300,6 +307,7 @@ targets:
       file: 'values.yaml'
       name: 'sample'
       key: '$.image.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'ghcr.io_epinio_epinio-server'
 `,
@@ -339,6 +347,7 @@ targets:
       file: 'values.yaml'
       name: 'sample'
       key: '$.images.ui.tag'
+      skippackaging: false
       versionincrement: ''
     sourceid: 'ghcr.io_epinio_epinio-ui'
 `},

